src/components/Initiatives/InitiativeDetailsPage/InitativeFilterDialog/InitiativeFilterDialogUtils.ts

```diff
@@ -0,0 +1,248 @@
+      acc[newKey] = filters[key];
+      return acc;
+    }, {});
+};
+
+export const getPredicateFilterFromFilters = (
+  filters: FilterShape,
+  oneOf: Record<string, boolean>,
+  filterDefinitions: FilterDefinitionWithPredicate<string>[],
+) => {
+  const allFilters = filterDefinitions.reduce<
+    Record<string, Predicate<string>>
+  >((acc, filter) => {
+    const predicateFilters = toPredicateFilters(filters, filter.name);
+
+    const filterOneOf = oneOf[filter.name] ?? true;
+
+    acc[filter.name] = (componentRef: string) => {
+      if (isEmpty(predicateFilters)) {
+        return true;
+      }
+
+      const results = Object.keys(predicateFilters)
+        .filter(id => predicateFilters[id])
+        .map(id =>
+          filter.generatePredicate(filter.filters[id].value)(componentRef),
+        );
+
+      return filterOneOf ? results.some(Boolean) : results.every(Boolean);
+    };
+
+    return acc;
+  }, {});
+
+  return combinePredicates(Object.values(allFilters));
+};
+
+const oneOfPrefix = 'oneOf__';
+
+export const toQueryParams = (
+  filters: FilterShape,
+  oneOf: Record<string, boolean>,
+  filterDefinitions: FilterDefinition[],
+) => {
+  const result: Record<string, string[]> = Object.entries(filters).reduce(
+    (acc, [key, value]) => {
+      if (!value) {
+        return acc;
+      }
+
+      const filter = filterDefinitions.find(filter =>
+        key.includes(filter.name),
+      );
+
+      if (!filter) {
+        return acc;
+      }
+
+      const filterEntry = key.replace(filter.name, '');
+
+      if (!acc[filter.name]) {
+        acc[filter.name] = [];
+      }
+      acc[filter.name].push(filterEntry);
+
+      return acc;
+    },
+    {} as Record<string, string[]>,
+  );
+
+  const oneOfResult: Record<string, boolean> = Object.entries(oneOf).reduce(
+    (acc, [key, value]) => {
+      if (!value) {
+        acc[`${oneOfPrefix}${key}`] = false;
+      }
+      return acc;
+    },
+    {} as Record<string, boolean>,
+  );
+
+  return {
+    ...oneOfResult,
+    ...result,
+  };
+};
+
+export const useFiltersFromQueryParams = (queryParams: string) => {
+  const searchParams = new URLSearchParams(queryParams);
+
+  const filters: Record<string, boolean> = {};
+  const oneOf: Record<string, boolean> = {};
+
+  searchParams.forEach((value, key) => {
+    if (key.startsWith(oneOfPrefix)) {
+      oneOf[key.replace(oneOfPrefix, '')] = value === 'true';
+    } else {
+      const filterName = Object.values(FilterDefinitionName).find(filterName =>
+        key.includes(filterName),
+      );
+      filters[`${filterName}${value}`] = true;
+    }
+  });
+
+  return { filters, oneOf };
+};
+
+const createRulePredicate = (
+  rule: string,
+  actionItems: InitiativeActionItem[],
+  pass: boolean,
+) => {
+  return (componentRef: AnyEntityRef) => {
+    const passed = !actionItems.some(
+      actionItem =>
+        actionItem.componentRef === componentRef &&
+        actionItem.rule.expression === rule,
+    );
```

Comment:
Not an issue but couldn't we just do a `.find` here if we only care that at least one is found?

Reply:
I don't think so -- the `pass` param of `createRulePredicate` determines the default behavior so we need to accommodate that. Also, `.some` and `.find` are the same efficiency (only diff is returning a bool vs the entry), so what's the benefit?

Reply:
In my head I was thinking `.some` iterated on all the things, but it is realistically yes the same and bails on at least one. All good.